src/utils.js

```diff
@@ -658,7 +658,9 @@ export default {
         } else if (def.type === 'hasOne') {
           if (localKey) {
             prop.get = function () {
-              return get(this, localKey) ? definition.getResource(relationName).get(get(this, localKey)) : undefined
+              let key = get(this, localKey);
+              let hasKey = isEmpty(key);
```

Comment:
What do you think about more simple expression?
`let hasDefKey = !!(defKey || defKey === 0)`